### PR TITLE
[AIRFLOW-1262] Allow configuration of email alert subject and body

### PR DESCRIPTION
**Pull request by alekstorm**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1262

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

Adds `email.subject` and `email.html_content` config values, named to mirror parameters to `EmailOperator`.

Note that a quirk of `configparser` requires Jinja2 delimiters to be doubled.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

- `tests.models:TaskInstanceTest.test_failure_email`

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -218,6 +218,14 @@ hide_paused_dags_by_default = False
 
 [email]
 email_backend = airflow.utils.email.send_email_smtp
+subject = Airflow alert: {{{{ti}}}}
```
Comment (bolkedebruin) [inline airflow/config_templates/default_airflow.cfg]: I think this is cruft to be honest. Why not use a template file?

Comment (alekstorm) [reply airflow/config_templates/default_airflow.cfg]: Sorry for the delay - I've reworked the PR to take `subject_template` and `html_content_template` config entries, each specifying a path to a template. Please let me know what you think.

**Comment by christophertull** [comment]
Would love to see this functionality implemented somehow!

**Comment by Fokko** [comment]
@alekstorm can you rebase onto master?

**Comment by alekstorm** [comment]
@Fokko Rebased and ready for another review.

**Review by Fokko — APPROVED** [review]
Thanks @alekstorm, looking good.

**Comment by Fokko** [comment]
Let me run the tests one more time: https://travis-ci.org/Fokko/incubator-airflow/builds/385586411

**Comment by kaxil** [comment]
@alekstorm Can you rebase with master and resolve conflicts?

**Comment by alekstorm** [comment]
Okay, rebased and got the tests to pass (they appear to be flaky). This is now the fourth time I've written this code - can you please get it merged? My team doesn't even need this feature anymore; this is spurred by a request on the mailing list.

**Comment by ashb** [comment]
Awesome, thanks.

**Comment by kaxil** [comment]
Thanks @alekstorm . Appreciate that.

**Comment by christophertull** [comment]
Thanks @alekstorm!

**Comment by ron819** [comment]
thanks @alekstorm

**Pull request merged**